src/injected/visualization/table-headers-formatter.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { DialogRenderer } from '../dialog-renderer';
+import { AssessmentVisualizationInstance } from '../frameCommunicators/html-element-axe-results-helper';
+import { FailureInstanceFormatter } from './failure-instance-formatter';
+import { DrawerConfiguration } from './formatter';
+
+export interface HeadingStyleConfiguration {
+    borderColor: string;
+    fontColor: string;
+}
+
+export interface StyleComputer {
+    getComputedStyle(elt: Element, pseudoElt?: string): CSSStyleDeclaration;
+}
+
+export class TableHeadersAttributeFormatter extends FailureInstanceFormatter {
+    private readonly cellColor = '#6600CC';
+    private readonly headerColor = '#0066CC';
+
+    constructor() {
+        super();
+    }
+
+    public getDialogRenderer(): DialogRenderer {
+        return null;
+    }
+
+    public getDrawerConfiguration(
+        element: HTMLElement,
+        data: AssessmentVisualizationInstance,
+    ): DrawerConfiguration {
+        const isHeader = this.isHeader(element);
+
+        const style = {
+            borderColor: isHeader ? this.headerColor : this.cellColor,
+            fontColor: '#FFFFFF',
+        };
+
+        const text = isHeader ? this.getTextForHeader(element) : this.getTextForCell(element);
+
+        return {
+            textBoxConfig: {
+                fontColor: style.fontColor,
+                text,
+                background: style.borderColor,
+            },
+            borderColor: style.borderColor,
+            outlineStyle: 'solid',
+            showVisualization: true,
+            textAlign: 'right',
+            failureBoxConfig: this.getFailureBoxConfig(data),
+        };
+    }
+
+    private isHeader(element: HTMLElement): boolean {
+        if (element.matches('th')) {
+            return true;
```

Comment:
Do we need to worry about elements like `<th role="somenonheader">`?

Reply:
(if not, would be nice to leave a comment explaining why)

Reply:
According to the text for Semantics > Table headers, and the W3 pages that we link from that requirement, using a `<th>` element is sufficient to code that element as a header. I can't find any mention of roles that would cause a problem for table headers. This check only affects the visual helper, which will show that the element is coded as a header regardless of its role.